app/models/qdm/basetypes/data_element.rb

```diff
@@ -89,6 +89,53 @@ def shift_facility_location_dates(facility_location, seconds)
       facility_location['locationPeriod'][:high] = (facility_location['locationPeriod'][:high].to_time + seconds).to_datetime
     end
 
+    # The necessary reason for this function is to avoid a problem when shifting
+    # past a year that is a leap year. February 29th dates are handled by moving
+    # back to the 28th in non leap years
+    def shift_years(year) 
+      # Iterate over fields
+      fields.keys.each do |field|
+        # Check if field is a DateTime
+        if send(field).is_a? DateTime
+          if send(field).month == 2 && send(field).day == 29 && !Date.leap?(year.to_i)
+            send(field + '=', send(field).change(year: year.to_i, day: 28))
+          else
+            send(field + '=', send(field).change(year: year.to_i))
+          end
+        end
+        # Check if field is an Interval
+        if (send(field).is_a? Interval) || (send(field).is_a? DataElement)
+          send(field + '=', send(field).shift_years(year))
+        end
+
+        # Special case for facility locations
```

Comment:
The if/elsif below should be handled polymorphically rather than in the base class:

To do this, remove the block below but extend `shift_years` on FacilityLocation to provide the functionality of `shift_facility_location_years` below after calling the dataElement super `shift_years`.  Then extend `shift_years` in Encounter Performed to include a loop over each `facilityLocation` that calls that function.